tensordict/prototype/tensorclass.py

```diff
@@ -102,7 +102,9 @@ def tensorclass(cls: T) -> T:
         fields=[("batch_size", torch.Size, field(default_factory=list))],
     )
 
-    EXPECTED_KEYS = set(datacls.__dataclass_fields__.keys())
+    EXPECTED_KEYS = {
+        k for k in datacls.__dataclass_fields__.keys() if k != "batch_size"
+    }
```

Comment:
If we don't exclude `batch_size` here then assinging to `.batch_size` attribute attempts to place the supplied value in the underlying tensordict under the key `"batch_size"`.